collections/src/main/scala/strawman/collection/immutable/ImmutableArray.scala

```diff
@@ -28,17 +28,17 @@ sealed abstract class ImmutableArray[+A]
   /** The tag of the element type */
   protected[this] def elemTag: ClassTag[A]
 
-  def iterableFactory: SeqFactory[ImmutableArray] = ImmutableArray.untagged
+  override def iterableFactory: SeqFactory[ImmutableArray] = ImmutableArray.untagged
 
   /** The wrapped mutable `Array` that backs this `ImmutableArray`. Any changes to this array will break
     * the expected immutability. */
   def unsafeArray: Array[A @uncheckedVariance]
   // uncheckedVariance should be safe: Array[A] for reference types A is covariant at the JVM level. Array[A] for
   // primitive types A can only be widened to Array[Any] which erases to Object.
 
-  protected[this] def fromSpecificIterable(coll: strawman.collection.Iterable[A]): ImmutableArray[A] = fromIterable(coll)
+  override protected[this] def fromSpecificIterable(coll: strawman.collection.Iterable[A]): ImmutableArray[A] = fromIterable(coll)
```

Comment:
Here what’s the difference with the base implementation?

Reply:
This is wrong and was equally wrong before. We need to build using `elemTag` here.